zio-kafka/src/main/scala/zio/kafka/consumer/internal/PartitionStreamControl.scala

```diff
@@ -54,8 +58,16 @@ object PartitionStreamControl {
   private[internal] def newPartitionStream(
     tp: TopicPartition,
     commandQueue: Queue[RunloopCommand],
-    diagnostics: Diagnostics
-  ): UIO[PartitionStreamControl] =
+    diagnostics: Diagnostics,
+    maxPollInterval: Duration,
+    onConsumeTimeout: UIO[Unit]
+  ): UIO[PartitionStreamControl] = {
+    val timeOutMessage = s"No records were polled for more than $maxPollInterval for topic partition $tp. " +
+      "Use ConsumerSettings.withMaxPollInterval to set a longer interval if processing a batch of records " +
+      "needs more time."
+    val consumeTimeout = new TimeoutException(timeOutMessage) with NoStackTrace
+    val onTimeout      = ZIO.logError(timeOutMessage) *> onConsumeTimeout
```

Comment:
If everything goes well, these things should never be instantiated, no?

```suggestion
    def timeOutMessage = s"No records were polled for more than $maxPollInterval for topic partition $tp. " +
      "Use ConsumerSettings.withMaxPollInterval to set a longer interval if processing a batch of records " +
      "needs more time."
    def consumeTimeout = new TimeoutException(timeOutMessage) with NoStackTrace
    def onTimeout      = ZIO.logError(timeOutMessage) *> onConsumeTimeout
```